tests/integration/security/ca_custom_root/trust_domain_alias_client_test.go

```diff
@@ -0,0 +1,182 @@
+// - When server side has trust domain which defined as trust domain alias, the mTLS connection should expect success
+// - When server side has trust domain which not defined as trust domain alias, the mTLS connection should expect fail
+func TestTrustDomainAliasClient(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.peer.trust-domain-validation-alias-client").
+		Run(func(ctx framework.TestContext) {
+			testNS := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "trust-domain-alias-client-validation",
+				Inject: true,
+			})
+
+			// Create testing certs using runtime namespace
+			//generateCerts(ctx, testNS.Name())
+			generateCerts(ctx, testNS.Name(), "foo", "server-naked-foo")
+			generateCerts(ctx, testNS.Name(), "bar", "server-naked-bar")
+			defer cleanCerts(ctx)
+
+			// Deploy 3 workloads:
+			// client: echo app with istio-proxy sidecar injected, holds default trust domain cluster.local
+			// serverNakedFoo: echo app without istio-proxy sidecar, holds custom trust domain trust-domain-foo
+			// serverNakedBar: echo app without istio-proxy sidecar, holds custom trust domain trust-domain-bar
+			var client, serverNakedFoo, serverNakedBar echo.Instance
+			echoboot.NewBuilder(ctx).
+				With(&client, echo.Config{
+					Namespace: testNS,
+					Service:   "client",
+				}).
+				With(&serverNakedFoo, echo.Config{
+					Namespace: testNS,
+					Service:   "server-naked-foo",
+					Subsets: []echo.SubsetConfig{
+						{
+							Annotations: echo.NewAnnotations().SetBool(echo.SidecarInject, false),
+						},
+					},
+					ServiceAccount: true,
+					Ports: []echo.Port{
+						{
+							Name:         HTTPS,
+							Protocol:     protocol.HTTPS,
+							ServicePort:  443,
+							InstancePort: 8443,
+							TLS:          true,
+						},
+					},
+					TLSSettings: &common.TLSSettings{
+						RootCert:   loadCert(t, "root-cert.pem"),
+						ClientCert: loadCert(t, "tmp/workload-foo-cert.pem"),
+						Key:        loadCert(t, "tmp/workload-foo-key.pem"),
+					},
+				}).
+				With(&serverNakedBar, echo.Config{
+					Namespace: testNS,
+					Service:   "server-naked-bar",
+					Subsets: []echo.SubsetConfig{
+						{
+							Annotations: echo.NewAnnotations().SetBool(echo.SidecarInject, false),
+						},
+					},
+					ServiceAccount: true,
+					Ports: []echo.Port{
+						{
+							Name:         HTTPS,
+							Protocol:     protocol.HTTPS,
+							ServicePort:  443,
+							InstancePort: 8443,
+							TLS:          true,
+						},
+					},
+					TLSSettings: &common.TLSSettings{
+						RootCert:   loadCert(t, "root-cert.pem"),
+						ClientCert: loadCert(t, "tmp/workload-bar-cert.pem"),
+						Key:        loadCert(t, "tmp/workload-bar-key.pem"),
+					},
+				}).
+				BuildOrFail(t)
+
+			ctx.Config().ApplyYAMLOrFail(ctx, testNS.Name(), fmt.Sprintf(POLICY))
+
+			verify := func(t *testing.T, src echo.Instance, dest echo.Instance, s scheme.Instance, allow bool) {
```

Comment:
allow/deny naming is inherited from validatoin test which make sense since it's testing server side behavior. Here you are not. (ALLOW/DENY is server side terminology)

Change to `success`

Reply:
fixed